### DISCUSSION TITLE
[Question] How does the app handle duplicates between domain blocklists?

### DISCUSSION THREAD

**Discussion by n8x0b7T** [celzero/rethink-app#908] [Q&A] [answered]
I briefly searched the issue tracker and code base to see if this has been addressed. If there are duplicates, could this affect performance for the resolver?

**Comment by ignoramous** [comment, marked as answer]
No it doesn't affect performance at all.

Domains name are stored in a *Succinct Radix Trie* (impl: [js](https://github.com/serverless-dns/trie), [go](https://github.com/celzero/gotrie)), which optimally handles duplicates and all.

**Comment by n8x0b7T** [comment]
Thanks for the info! Good to know.
